hyperspy/signal_tools.py

```diff
@@ -710,12 +739,22 @@ def plot_histogram(self, max_num_bins=250):
         if self._vmin == self._vmax:
             return
         data = self._get_data()
+        # masked data outside vmin/vmax
+        data = np.ma.masked_outside(data, self._vmin, self._vmax).compressed()
 
-        # "auto" returns max('sturges', 'fd') which is what we want
-        n_bins = len(np.histogram_bin_edges(data, bins="auto"))
-        # Cap at max_num_bins to avoid memory errors when
-        # the number of bins is very large
-        self.bins = min(n_bins, max_num_bins)
+        # Sturges rule
+        sturges_bin_width = data.ptp() / (np.log2(data.size) + 1.0)
+
+        iqr = np.subtract(*np.percentile(data, [75, 25]))
+        fd_bin_width = 2.0 * iqr * data.size ** (-1.0 / 3.0)
+
+        if fd_bin_width > 0:
+            bin_width = min(fd_bin_width, sturges_bin_width)
+        else:
+            # limited variance: fd_bin_width may be zero
+            bin_width = sturges_bin_width
+
+        self.bins = min(int(np.ceil(data.ptp() / bin_width)), 250)
```

Comment:
Replace `250` with `max_num_bins`

Reply:
Oh yes, I was too quick with copy and paste!